projects/plugins/super-cache/wp-cache.php

```diff
@@ -3271,14 +3275,39 @@ function wp_cron_preload_cache() {
 						fclose( $fp );
 					}
 				}
+
+				if (
+					$preload_more_taxonomies === false &&
+					count( $details ) !== 1 &&
+					$details[0] !== ''
+				) {
+					$preload_more_taxonomies = true;
+				}
+			}
+
+			if ( $preload_more_taxonomies === true ) {
+				if ( defined( 'DOING_CRON' ) ) {
+					wp_cache_debug( 'wp_cron_preload_cache: scheduling the next taxonomy preload in 3 seconds.' );
+					wp_schedule_single_event( time() + 3, 'wp_cache_preload_hook' );
+					wp_delete_file( $mutex );
```

Comment:
Is it worth taking this opportunity to move the scheduling and mutex stuff into its own function, to separate it from the work of preloading - to make the code easier to maintain in future?